packages/amplify-provider-awscloudformation/index.js

```diff
@@ -43,8 +44,15 @@ function buildResources(context, category, resourceName) {
   return resourceBuilder.run(context, category, resourceName);
 }
 
-function getConfiguredAWSClient(context) {
-  return aws.configureWithCreds(context);
+function getConfiguredAWSClient(context, category, action) {
```

Comment:
The con for this proposed refactoring is that `getCurrentConfig` is an expensive method call, it potentially reads files twice from the disk, so calling it twice is not economical. Considering that we are having performance concerns already, this might not be the best choice at the moment. 
But I do think there's room for improvements for both code clarity and performance in the module amplify-provider-awscloudformation, as we discussed offline.